addons/account/models/account_move.py

```diff
@@ -1528,6 +1528,77 @@ def _prepare_tax_line_for_taxes_computation(self, tax_line):
             sign=self.direction_sign,
         )
 
+    def _prepare_non_deductible_base_line_for_taxes_computation(self, non_deductible_line):
+        """ Convert an account.move.line having display_type='non_deductible' into a base line for the taxes computation.
+
+        :param non_deductible_line: An account.move.line.
+        :return: A base line returned by '_prepare_base_line_for_taxes_computation'.
+        """
+        self.ensure_one()
+        sign = self.direction_sign
+        rate = self.invoice_currency_rate
+        special_type = 'non_deductible' if non_deductible_line.display_type == 'non_deductible_product' else False
+        return self.env['account.tax']._prepare_base_line_for_taxes_computation(
+            non_deductible_line,
+            price_unit=sign * non_deductible_line.amount_currency,
+            quantity=1.0,
+            sign=sign,
+            special_mode='total_excluded',
+            special_type=special_type,
+
+            is_refund=self.move_type in ('out_refund', 'in_refund'),
+            rate=rate,
+        )
+
+    def _prepare_non_deductible_base_lines_for_taxes_computation_from_base_lines(self, base_lines):
+        """ Anticipate the non deductible lines to be generated from the base lines passed as parameter.
+        When the record is in draft (not saved), the accounting items are not there so we can't
+        call '_prepare_non_deductible_base_line_for_taxes_computation'.
+
+        :param base_lines: The base lines generated by '_prepare_product_base_line_for_taxes_computation'.
+        :return: A list of base lines representing the non deductible lines.
+        """
+        self.ensure_one()
+        non_deductible_product_lines = base_lines.filtered(lambda line: line.display_type == 'product' and float_compare(line.deductible_amount, 100, precision_digits=2))
+        if not non_deductible_product_lines:
+            return []
+
+        sign = self.direction_sign
+        rate = self.invoice_currency_rate
+
+        non_deductible_lines_base_total_currency = 0.0
+        non_deductible_lines = []
+        for line in non_deductible_product_lines:
+            percentage = (1 - line.deductible_amount / 100)
+            non_deductible_subtotal = line.currency_id.round(line.price_subtotal * percentage)
+            non_deductible_base_currency = line.company_currency_id.round(sign * non_deductible_subtotal / rate) if rate else 0.0
+            non_deductible_lines_base_total_currency += non_deductible_base_currency
+
+            non_deductible_lines += [
+                self.env['account.tax']._prepare_base_line_for_taxes_computation(
+                    self.env['account.move.line'],
```

Comment:
```suggestion
                    None,
```
I think it works :thinking: